app/javascript/components/layer.jsx

```diff
@@ -21,27 +24,26 @@ export function Layer({ scope, layer, isOwner, layerId, activeTool, activeColor
 
       received(data) {
         // Called when there's incoming data on the websocket for this channel
-        // TODO: generalize to receive diff and add the proper type of element to the layer
-        newLayer.addChild(new Paper.PointText({
-          point: [50, 50],
-          content: data,
-          fillColor: (isOwner) ? 'red' : 'blue',
-          fontSize: 25
-        }));
+        // TODO: Check to see if this diff has already been applied on this client
+        // TODO: Verify seq number is in order
+        newLayer.importJSON(data.diff)
       }
     });
   }
 
-  // Should be called after a user creates a diff
+  // Should be called after a user creates a diff. Diff should be a PaperJS object
   const transmitDiff = (diff) => {
-    console.log("should transmit diff");
-    // TODO: implement logic to send diff back to server over `layer_channel_${layerId}`
+    // TODO: This should be called for things like erase and text too
+    // TODO: This isn't running the first time through. Why is it null the first time?
+    if (layerChannel !== null) {
```

Comment:
Yea weird that its null the first time, I will do some digging and see if I can figure it out.